### ISSUE TITLE
Datepicker: Panel not responsive

### ISSUE THREAD

**Issue by miedrl** [primefaces/primefaces#8381]
**Describe the defect**
The overlay panel of the datepicker component is not responsive and thus overflows small screens and can't be used to its full extent.

**Environment:**
 - PF Version: _10.0.0_

**To Reproduce**
1. Go to the [PrimeFaces datepicker showcase](http://primefaces.org/showcase/ui/input/datepicker/datePicker.xhtml?jfwid=d28da).
2. Reduce the screen width (e.g. to 320px).
3. Click on one of the datepicker components to open its overlay panel.
4. See error > overlay panel overflows the screen.

**Expected behavior**
The overlay panel on the datepicker component should adapt its width to the given screen width to be responsive. This is important to fulfil [WCAG accessibility success criterion 1.4.10](https://www.w3.org/WAI/WCAG21/Understanding/reflow.html). The responsiveness should be guaranteed no matter what attributes are set (e.g. monthNavigator, yearNavigator and showButtonBar).

**Comment by jepsar** [comment]
For me the showcase is broken when I use it with my mobile. Somehow the inline picker is displayed statically and blocking most of the screen.

@miedrl can you try the "Touch UI" example?

**Comment by miedrl** [comment]
@jepsar When I open the showcase on my mobile, I get the same problem as you. I checked the "Touch UI" example on my computer using developer tools to reduce the screen down to 320px. The panel overflows the screen here too. Here is a screenshot of the result:

![grafik](https://user-images.githubusercontent.com/77431692/152304743-f5cd2832-0973-4ea1-98bb-ca8a95c7a613.png)

**Comment by jepsar** [comment]
You should check on a touch enabled device, it will auto detect touchability.

<img src="https://user-images.githubusercontent.com/7500178/152307716-a45d784f-0eb8-4f02-a01e-a66c82a6c24a.jpg" width="420"/>

**Comment by miedrl** [comment]
@jepsar I tried again with my laptop (which has a touch display). The result for touching the component is the same as when clicking on it.

**Comment by jepsar** [comment]
Can you go to the showcase and try this in your JS console?
```js
'ontouchstart' in window || window.navigator.msMaxTouchPoints || PrimeFaces.env.mobile;
```
It should return `true` if your device is a touch device according to PrimeFaces.

**Comment by miedrl** [comment]
@jepsar I tried and it returned false so it seems that my device is not a touch device according to PrimeFaces. Sorry that I could not help you here.

**Comment by jepsar** [comment]
That check could be improved though, see for example https://javascript.plainenglish.io/how-to-detect-a-touch-screen-device-using-javascript-68a68360375a

https://hacks.mozilla.org/2013/04/detecting-touch-its-the-why-not-the-how/

I think date picker should switch to full screen based on screen size and not on touchable. Not sure if it does not already (did not investigate that yet).

**Comment by melloware** [comment]
It does not check screen size I added this feature in PF11 to autodetect touchable but not screen size.

https://github.com/primefaces/primefaces/issues/7126

**Comment by melloware** [comment]
This is how touch punch determines touch successfully...

```javascript
touch = ( 'ontouchstart' in document
   	|| 'ontouchstart' in window
   	|| window.TouchEvent
   	|| (window.DocumentTouch && document instanceof DocumentTouch)
   	|| navigator.maxTouchPoints > 0
   	|| navigator.msMaxTouchPoints > 0
  );
```

**Comment by melloware** [comment]
@jepsar i thnk the problem with screen size is who determines the size at which to switch it.  They can always use the `touchUI=#{myvaraible}"` and update the value based on screen size changes right?

**Comment by melloware** [comment]
There was this ticket that was long ago rejected: https://github.com/primefaces/primefaces/issues/1281

**Comment by jepsar** [comment]
True, but that would mean at least one roundtrip to get the JS variable available as EL. A global setting "default to touch UI at maximum x pixels" would be more convenient I think. Also, the touch UI of the date picker won't be very useful on a huge touchable screen.

Or `touchUI` should be a JS function instead of a boolean.

**Comment by jepsar** [comment]
... or we could convert `touchUI` to a string where `true` and `false` will be the same as current implementation, anything else will be treated as a JS function which should return a boolean.

**Comment by melloware** [comment]
Not a bad idea.

**Comment by melloware** [comment]
This took a little bit more than just a change in function it has to listen to `window.resize` events.  Please review.
